src/lfortran/semantics/ast_body_visitor.cpp

```
@@ -1593,29 +1584,29 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
 
                 for (size_t i = 0; i < x.n_labels; ++i) {
                     if (!AST::is_a<AST::Num_t>(*x.m_labels[i])) {
-                        throw SemanticError("Can only `GOTO` integer labels",
+                        throw SemanticError("Only integer labels are supported in GOTO.",
                             x.base.base.loc);
                     } else {
                         auto l = AST::down_cast<AST::Num_t>(x.m_labels[i]); // l->m_n gets the target -> if l->m_n == (i+1) ...
                         Vec<ASR::stmt_t*> body;
                         body.reserve(al, 1);
                         body.push_back(al, ASRUtils::STMT(ASR::make_GoTo_t(al, x.base.base.loc, l->m_n)));
+                        if (i == 0) {
+                            def_body.push_back(al, ASRUtils::STMT(ASR::make_GoTo_t(al, x.base.base.loc, l->m_n)));
+                        }
```

Is that the Fortran semantics that if no label matches at runtime, that it jumps to the first one?

Or should we rather give a runtime error?

---

I might misread something when I looked for documentation. I remembered seeing this, but when I checked now, 

- https://docs.oracle.com/cd/E19957-01/805-4939/6j4m0vn9l/index.html
- https://www.ibm.com/docs/en/openxl-fortran-aix/17.1.0?topic=attributes-go-computed
- https://www.intel.com/content/www/us/en/develop/documentation/fortran-compiler-oneapi-dev-guide-and-reference/top/language-reference/a-to-z-reference/g-1/goto-computed.html

They all say that it should be a no-op when the computed value is out of range.

---

We can check what gfortran does. If it is undefined, then let's return a runtime error.

---

GFortran does not throw errors but simply goes ahead to the next statement (making the goto a no-op).

---

Changed out-of-range values to no-op as the listed documentation.